Source/Fuse.Models/ModelJavaScript.uno

```diff
@@ -126,22 +177,38 @@ namespace Fuse.Models
 			}
 			
 			//app-level model does not have a nametable otherwise migration would not be possible
-			var js = new ModelJavaScript(null, modulePath, previewStateId);
-			return js;
+			var md = new ModelData
+			{
+				ModulePath = modulePath
+			};
+			
+			return new ModelJavaScript(md, previewStateId);
 		}
 		
 		string _modulePath;
-		internal ModelJavaScript(NameTable nt, string modulePath, string previewStateId)
-			: base(nt)
+		IExpression _args;
+		bool _hasArgs;
+
+		private ModelJavaScript(ModelData md, string previewStateId = null)
+			: base(md.NameTable)
 		{
 			_previewStateModelId = previewStateId;
-			_modulePath = modulePath;
+			_modulePath = md.ModulePath;
+			_args = md.Arguments;
+			_hasArgs = md.HasArguments;
 			FileName = "(model-script)";
-			SetupModel();
 		}
 		
 		protected override void OnRooted()
 		{
+			// TODO: Doing anything prior to base.OnRooted is questionable.
```

Comment:
Can you address this TODO?  Perhaps make a new virtual function in the base class called by OnRooted.